tests/cnf/core/network/metallb/tests/bgp-unnumbered.go

```diff
@@ -0,0 +1,282 @@
+				srIovInterfacesUnderTest[0])
+
+			By("Collecting the frrk8sPod Pod list for worker nodes in list cnfWorkerNodeList")
+			frrk8sPods = verifyAndCreateFRRk8sPodList()
+
+			By("Creating BFD profile.")
+			createBFDProfileAndVerifyIfItsReady(frrk8sPods)
+
+			By("Creating host-device NAD for external FRR pod")
+			_, err = define.HostDeviceNad(APIClient, frrconfig.ExternalMacVlanNADName, srIovInterfacesUnderTest[0],
+				tsparams.TestNamespaceName, nad.IPAMStatic())
+			Expect(err).ToNot(HaveOccurred(), "Failed to create a network-attachment-definition")
+
+			By("Create a single IPAddressPool")
+			ipAddressPool := createIPAddressPool("ipaddresspool", ipAddressPoolRange)
+
+			By("Creating a MetalLB service")
+			setupMetalLbService(tsparams.MetallbServiceName, netparam.IPV4Family, tsparams.LabelValue1,
+				ipAddressPool, "Cluster")
+
+			By(fmt.Sprintf(
+				"Creating a BGPAdvertisement with the nodeSelector to %s", workerNodeList[0].Definition.Name))
+			worker0NodeLabel := []metav1.LabelSelector{
+				{MatchLabels: map[string]string{corev1.LabelHostname: workerNodeList[0].Definition.Name}},
+			}
+
+			setupBgpAdvertisement("bgpadvertisment", tsparams.CustomCommunity, ipAddressPool.Definition.Name,
+				200, []string{tsparams.BgpPeerName1}, worker0NodeLabel)
+		})
+
+		AfterAll(func() {
+			By("Remove custom MetalLB test label from nodes")
+			removeNodeLabel(workerNodeList, metalLbTestsLabel)
+
+			By(fmt.Sprintf("Disabling ethernet interface %s on worker node %s",
+				srIovInterfacesUnderTest[0], workerNodeList[0].Definition.Name))
+			ethIntWorker0Policy := nmstate.NewPolicyBuilder(APIClient, nodeNetConfigPolicyName, NetConfig.WorkerLabelMap).
+				WithAbsentInterface(srIovInterfacesUnderTest[0])
+			err := netnmstate.UpdatePolicyAndWaitUntilItsAvailable(netparam.DefaultTimeout, ethIntWorker0Policy)
+			Expect(err).ToNot(HaveOccurred(), "Failed to update NMState network policy")
+
+			By("Removing NMState policies")
+			err = nmstate.CleanAllNMStatePolicies(APIClient)
+			Expect(err).ToNot(HaveOccurred(), "Failed to remove all NMState policies")
+
+			By("Clean MetalLB operator and test namespaces")
+			resetOperatorAndTestNS()
+		})
+
+		AfterEach(func() {
+			By("Reset MetalLB operator and test namespaces for next test.")
+			resetTestNSBetweenTestCases()
+		})
+
+		It("Verify iBGP peering established with BGP Unnumbered",
+			reportxml.ID("80393"), func() {
+				By("Verifying that the frrk8sPod deployment is in Ready state and create a list of the pods on " +
+					"worker nodes.")
+
+				By(fmt.Sprintf("Verify the frrk8s pod's link local IP address from interface %s on worker node %s",
+					srIovInterfacesUnderTest[0], workerNodeList[0].Definition.Name))
+				linkLocalAddress := getLinkLocalAddress(frrk8sPods[0], srIovInterfacesUnderTest[0], tsparams.FRRContainerName)
+
+				By("Create a frr config-map")
+				frrConfigMap := createConfigMapWithUnnumbered(tsparams.LocalBGPASN, tsparams.LocalBGPASN,
```

Comment:
Is this the frr config for the remote peer? that config should not need the linkLocalAddress of the metallb peer.
If that is required due to BFD, then we have to investigate. 
We would need  probably a test with BFD, and a same test without BFD.

Reply:
It is needed for the bfd configuration. It is for the issue are having created the external frr pod on the same cluster. The workaround is either starting the bfd process with an init or restarting the bfd process after bgp is configured. It is easier for us just to add the linklocal peer to the bfd configuration.

Reply:
Removed the workaround. Now now IP addresses are configured on any of the interfaces as well as the bfd configuration